### ISSUE TITLE
[data grid][question] Is it possible to make the GridToolbarQuickFilter textbox outlined?

### ISSUE THREAD

**Issue by elafito** [mui/mui-x#8544]
### The problem in depth 🔍

Is there a way to make the GridToolbarQuickFilter textbox outlined? I have tried many things with no success.
![image](https://user-images.githubusercontent.com/10082256/230623970-77aae68a-de0e-4812-946f-38e58faca5f9.png)

Preferably from theme.tsx, but any other suggestions are welcome.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

  System:
    OS: Windows 10 10.0.19045
  Binaries:
    Node: 18.12.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.19.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: Not Found
    Edge: Spartan (44.19041.1266.0), Chromium (104.0.1293.63)
  npmPackages:
    @emotion/react: ^11.10.6 => 11.10.6
    @emotion/styled: ^11.10.6 => 11.10.6
    @mui/base:  5.0.0-alpha.123
    @mui/core-downloads-tracker:  5.11.15
    @mui/icons-material: ^5.11.11 => 5.11.11
    @mui/material: ^5.11.15 => 5.11.15
    @mui/private-theming:  5.11.13
    @mui/styled-engine:  5.11.11
    @mui/system:  5.11.15
    @mui/types:  7.2.3
    @mui/utils:  5.11.13
    @mui/x-data-grid: ^6.0.4 => 6.0.4
    @types/react: ^18.0.32 => 18.0.32
    react: ^18.2.0 => 18.2.0
    react-dom: ^18.2.0 => 18.2.0
    typescript: ^4.9.5 => 4.9.5
</details>

**Comment by yaredtsy** [comment]
hey @elafito 

yes, you can, everything is customizable. you can refer to the documentation about customizing components at [here](https://mui.com/x/react-data-grid/components/).

you can pass `props` to components by using `componetsProps` in v5 or `slotsProps` in v6.  you can pass properties to  `quickFilter` by overding the `quickFilterProps`

```jsx
 <DataGrid
        ....
        components={{ Toolbar: GridToolbar }}
        componentsProps={{
          toolbar: {
            showQuickFilter: true,
            quickFilterProps: { debounceMs: 500, variant: "outlined" }
          }
        }}
      />
```
Demo: https://codesandbox.io/s/silly-christian-gjcvum?file=/demo.tsx

**Comment by yaredtsy** [comment]
hey @Venkat-Sparity 

`GridToolBar` component accepts props. You need to pass `props.quickFilterProps` to   `GridToolbarQuickFilter` .

```jsx
const GridToolBar = (props: GridToolbarProps) => {
  console.log(props);
  return (
    <Box>
      <GridToolbarContainer >
        <GridToolbarQuickFilter {...props.quickFilterProps} />
      </GridToolbarContainer>
    </Box>
  );
};
```

**Comment by michelengelen** [comment]
Yes, this is possible, and since v8 it is the default. #15909 changed the quick filter input to `variant="outlined"` with `size="small"`. The change shipped in [v8.0.0](https://github.com/mui/mui-x/releases/tag/v8.0.0). Since v8, the grid also respects `theme.components.MuiTextField.defaultProps.variant`, so you can control the variant from your theme, as you preferred.

On v6 and v7, use the workaround from [this comment](https://github.com/mui/mui-x/issues/8544#issuecomment-1500816103): pass `quickFilterProps: { variant: 'outlined' }` through the toolbar slot props.

The question is answered, so I will close this issue. Comment here if you need more help.
